In [1]:
import pandas as pd
house = pd.read_csv("Bangalore.csv")
house.head()
house.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6207 entries, 0 to 6206
Data columns (total 40 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Price                6207 non-null   int64 
 1   Area                 6207 non-null   int64 
 2   Location             6207 non-null   object
 3   No. of Bedrooms      6207 non-null   int64 
 4   Resale               6207 non-null   int64 
 5   MaintenanceStaff     6207 non-null   int64 
 6   Gymnasium            6207 non-null   int64 
 7   SwimmingPool         6207 non-null   int64 
 8   LandscapedGardens    6207 non-null   int64 
 9   JoggingTrack         6207 non-null   int64 
 10  RainWaterHarvesting  6207 non-null   int64 
 11  IndoorGames          6207 non-null   int64 
 12  ShoppingMall         6207 non-null   int64 
 13  Intercom             6207 non-null   int64 
 14  SportsFacility       6207 non-null   int64 
 15  ATM                  6207 non-null   int64 
 16  ClubHo

We have no empty column

In [2]:
house.isnull().sum()

Price                  0
Area                   0
Location               0
No. of Bedrooms        0
Resale                 0
MaintenanceStaff       0
Gymnasium              0
SwimmingPool           0
LandscapedGardens      0
JoggingTrack           0
RainWaterHarvesting    0
IndoorGames            0
ShoppingMall           0
Intercom               0
SportsFacility         0
ATM                    0
ClubHouse              0
School                 0
24X7Security           0
PowerBackup            0
CarParking             0
StaffQuarter           0
Cafeteria              0
MultipurposeRoom       0
Hospital               0
WashingMachine         0
Gasconnection          0
AC                     0
Wifi                   0
Children'splayarea     0
LiftAvailable          0
BED                    0
VaastuCompliant        0
Microwave              0
GolfCourse             0
TV                     0
DiningTable            0
Sofa                   0
Wardrobe               0
Refrigerator           0


In [3]:
house.Location .unique()

array(['JP Nagar Phase 1', 'Dasarahalli on Tumkur Road',
       'Kannur on Thanisandra Main Road', 'Doddanekundi', 'Kengeri',
       'Horamavu', 'Thanisandra', 'Ramamurthy Nagar',
       'Whitefield Hope Farm Junction', 'Electronic City Phase 1',
       'Yelahanka', 'Anjanapura', 'Jalahalli', 'Kasavanahalli',
       'Bommasandra', 'Bellandur', 'RR Nagar', 'Begur', 'Hosa Road',
       'Sahakar Nagar', 'Kadugodi', 'Jakkur', 'Jigani', 'Krishnarajapura',
       'Brookefield', 'Banashankari', 'Nelamangala', 'Attibele',
       'Banaswadi', 'Kodigehalli', 'ITPL', 'Uttarahalli Hobli',
       'Chikkagubbi on Hennur Main Road', 'Varthur', 'Vidyaranyapura',
       'Electronic City Phase 2', 'J. P. Nagar', 'K. Chudahalli',
       'Narayanaghatta', 'Anekal City', 'Sarjapur', 'Koramangala',
       'Hebbal', 'Budigere Cross', 'Bommanahalli', 'Electronics City',
       'Chikkalasandra', 'Kogilu', 'Nayandahalli', 'Bilekahalli',
       'Muneshwara Nagar', 'Junnasandra',
       'Narayanapura on Hennur Ma

In [4]:
len(house.Location .unique())

302

Increased number of location during hot encoding could give dimentionality issue but here 302 is fine to go with

In [5]:
house.Location = house.Location.apply(lambda x : x.strip())
locStats = house.groupby('Location')['Location'].agg('count').sort_values(ascending = False)
locStats

Location
Electronic City Phase 2    232
RR Nagar                   217
Begur                      186
Varthur                    168
Kumaraswamy Layout         154
                          ... 
Off Cunningham Road          1
OMBR Layout                  1
Nirman Layout                1
Bannerghatta Main Road       1
Murugesh Palya               1
Name: Location, Length: 302, dtype: int64

OUTLIER DETECTION : data points that represent the extreme variations in data and can cause errors in prediction here we are assuming that no room can be less than 300sqft  here we have 6 such cases

In [13]:
house[house.Area/house['No. of Bedrooms'] < 300]

Price  Area                         Location  No. of Bedrooms  \
0     30000000  3340                 JP Nagar Phase 1                4   
1      7888000  1045       Dasarahalli on Tumkur Road                2   
2      4866000  1179  Kannur on Thanisandra Main Road                2   
3      8358000  1675                     Doddanekundi                3   
4      6845000  1670                          Kengeri                3   
...        ...   ...                              ...              ...   
6202   5364000   590                       Chandapura                1   
6203   8716000  1179                    Kasavanahalli                2   
6204   7373000  1143                    Kasavanahalli                2   
6205   4985000  1680                    Kasavanahalli                3   
6206  10900000  1162                    Kasavanahalli                2   

      Resale  MaintenanceStaff  Gymnasium  SwimmingPool  LandscapedGardens  \
0          0                 1          1             1                  1   
1          0                 0          1             1                  1   
2          0                 0          1             1                  1   
3          0                 0          0             0                  0   
4          0                 1          1             1                  1   
...      ...               ...        ...           ...                ...   
6202       0                 9          9             9                  9   
6203       0                 9          9             9                  9   
6204       0                 9          9             9                  9   
6205       0                 9          9             9                  9   
6206       0                 9          9             9                  9   

      JoggingTrack  ...  LiftAvailable  BED  VaastuCompliant  Microwave  \
0                1  ...              1    0                0          0   
1                1  ...              1    0                1          0   
2                1  ...              1    0                0          0   
3                0  ...              1    0                0          0   
4                1  ...              1    0                0          0   
...            ...  ...            ...  ...              ...        ...   
6202             9  ...              9    9                9          9   
6203             9  ...              9    9                9          9   
6204             9  ...              9    9                9          9   
6205             9  ...              9    9                9          9   
6206             9  ...              9    9                9          9   

      GolfCourse  TV  DiningTable  Sofa  Wardrobe  Refrigerator  
0              0   0            0     0         0             0  
1              0   0            0     0         0             0  
2              0   0            0     0         0             0  
3              0   0            0     0         0             0  
4              0   0            0     0         0             0  
...          ...  ..          ...   ...       ...           ...  
6202           9   9            9     9         9             9  
6203           9   9            9     9         9             9  
6204           9   9            9     9         9             9  
6205           9   9            9     9         9             9  
6206           9   9            9     9         9             9  

[6207 rows x 40 columns]

In [12]:
house_area_bhk = house[~(house.Area/house['No. of Bedrooms'] < 300)]
house_area_bhk.shape

(6201, 40)